src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -185,6 +185,85 @@ dumpBabelGUCs(Archive *fout)
 	destroyPQExpBuffer(qry);
 }
 
+/*
+ * dumpBabelRestoreChecks:
+ * Dumps Babelfish specific pre-checks which get executed at the
+ * beginning of restore to validate if restore can be performed
+ * or not.
+ */
+void
+dumpBabelRestoreChecks(Archive *fout)
+{
+	PGresult	*res;
+	char		*source_server_version;
+	char		*source_migration_mode;
+	PQExpBuffer	qry;
+
+	/* Skip if not Babelfish database or binary upgrade */
+	if (!isBabelfishDatabase(fout) || fout->dopt->binary_upgrade)
+		return;
+
+	/*
+	 * Cross version Babelfish dump/restore is not yet supported so
+	 * store the current server's version in the below procedure and
+	 * add logic to fail the restore if the target server version
+	 * differs from source server version.
+	 */
+	qry = createPQExpBuffer();
+	res = ExecuteSqlQueryForSingleRow(fout, "SHOW server_version");
```

Comment:
do we want to block even patch level difference?

Reply:
I have not seen the cases where we have made any catalog changes between patch versions so I don't think it's necessary.

Reply:
but that is what we are doing, any difference at all between source and target will lead to failure (even at patch level)